PULL_REQUEST_TEMPLATE.md

```diff
@@ -2,48 +2,68 @@
 
-## Description
+## Motivation and Context
+
+<!--- Why is this change required? What problem does it solve? -->
+
+Please include relevant motivation and context. List any dependencies that are required for this change.
 
-<!--- Describe your changes in detail. Do not repeat the Issue description. -->
+## Implementation
 
-## Related Issue
+Please include a general summary of the change. 
+
+## Related Feature/Issue Jira tickets [optional]
 
-<!--- This project only accepts pull requests related to open issues -->
-<!--- If suggesting a new feature or change, please discuss it in an issue first -->
 <!--- If fixing a bug, there should be an issue describing it with steps to reproduce -->
 <!--- Please link to the issue here: -->
 
-## Motivation and Context
-
-<!--- Why is this change required? What problem does it solve? -->
+Please, just in case more than one feature/issue is included in the pull request, include all the Jira tickets kind of feature/issue. 
 
 ## How Has This Been Tested
 
 <!--- Please describe in detail how you tested your changes. -->
 <!--- Include details of your testing environment, and the tests you ran to -->
 <!--- see how your change affects other areas of the code, etc. -->
 
+Please describe the tests that you ran to verify your changes. Please also note any relevant details for your test configuration.
+
+- [ ] Test A
+- [ ] Test B
+
+
 ## Types of changes
 
 <!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
 
 - [ ] Docs change / refactoring / dependency upgrade
 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [ ] New feature (non-breaking change which adds functionality)
+- [ ] New release version 
 - [ ] Breaking change (fix or feature that would cause existing functionality to change)
 
+## Attachments
+
+<!--- Please add all the needed attachments -->
+
+Please include the relevant attachments.
+
+
 ## Checklist
 
 <!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
 <!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
 
 - [ ] I have read the **CONTRIBUTING** document
 - [ ] My code follows the code style of this project
-- [ ] My change requires a change to the documentation 
-- [ ] I have updated the documentation accordingly (README.md for each UI component)
-- [ ] I have added tests to cover my changes (at least 1 spec for each UI component with the same coverage as the master branch)
-- [ ] All new and existing tests passed
+- [ ] I have performed a self-review of my own code
+- [ ] I have commented my code, particularly in hard-to-understand areas
+- [ ] I have made corresponding changes to the documentation
+- [ ] There are **unit tests** for new/changed code (or a good explanation why this was not possible) and all of them passed
+- [ ] I have added tests to cover my changes (in REST API, at least 1 test for each endpoint with the same coverage as the master branch)
```

Comment:
I would merge the 2 options for the tests into a single one considering the scope of this project. It doesn't have REST API, so there aren't endpoints to cover. Moreover, the test project is not at unit level, so it can be better classified as a component test project.